lib/much-rails/action/change_action.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+require "much-rails/action"
+require "much-rails/change_action_result"
+require "much-rails/config"
+require "much-rails/plugin"
+
+# MuchRails::Action::ChangeAction defines the common behaviors for all view
+# action classes that change (ie save or destroy) records.
+module MuchRails; end
+module MuchRails::Action; end
+module MuchRails::Action::ChangeAction
+  Error = Class.new(StandardError)
+
+  include MuchRails::Plugin
+
+  plugin_included do
+    include MuchRails::Config
+    include MuchRails::Action
+
+    add_config :much_rails_action_change_action
+
+    on_after_call do
+      if any_unextracted_change_result_validation_errors?
+        raise(
+          MuchRails::Action::ActionError,
+          unhandled_change_result_action_error_message,
+          # NOTE: It may be better to kick the can down the road by doing:
+          # `cause: save_result.exception` instead?
+          unhandled_change_result_action_error_backtrace)
+      end
+    end
+  end
+
+  plugin_class_methods do
+    def change_result(&block)
+      much_rails_action_change_action_config.change_result_block = block
+    end
+  end
+
+  plugin_instance_methods do
+    def change_result
+      @change_result ||=
+        begin
+          unless self.class.much_rails_action_change_action_config.change_result_block
+            raise(Error, "A `change_result` block must be defined.")
```

Comment:
This may be future coding, but since you didn't also add SaveAction and DestroyAction here, this may need to change either here or in the future b/c the user won't actually set a `change_result` block. They will either set a `save_result` block or a `destroy_result` block:

```ruby
            raise(Error, "A change result block (e.g. `save_result` or `destroy_result` must be defined.")
```

Another approach might be to leave this alone and have the Save/Destroy Actions raise their own specific errors with specific error messages. Thoughts?